src/pluto/components/mismatch_report.py

```diff
@@ -0,0 +1,241 @@
+import streamlit as st
+import pandas as pd
+import plotly.graph_objects as go
+from src.constants import COLOR_SCHEME
+import pdb
+
+
+class MismatchReport:
+    def __init__(self, data, v1, v2, v3, condo, mapped):
+        self.df_mismatch = data
+        self.v1 = v1
+        self.v2 = v2
+        self.v3 = v3
+        self.condo = condo
+        self.mapped = mapped
+
+    def __call__(self):
+        df = self.filter_by_options()
+        v1v2 = self.filter_by_version(df, f"{self.v1} - {self.v2}")
+        v2v3 = self.filter_by_version(df, f"{self.v2} - {self.v3}")
```

Comment:
Can these go in the init too?

Reply:
My philosophy is usually that the init should not have a ton of business logic - now that I know how this property thing works lol, my instinct to make this function smaller would be to just stick each of these in its own property function instead. might be overkill though, its got a certain kind of procedural logic atm that maybe makes it easier to read

Reply:
Ok that makes about the init. I personally feel that if "setting up" the object requires logic then that logic should go in the init, but this way is ok too. I imagined the call as a more pure "give me the content" method but I'll have plenty of time to make that tweak going forward if I want to.
What do you mean about readable "procedural" logic?